geode-serialization/src/main/java/org/apache/geode/internal/serialization/filter/UnableToSetSerialFilterException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.serialization.filter;
+
+/**
+ * Checked exception thrown when there's a failure using Java's ObjectInputFilter. All uses of this
+ * exception are caught and rethrown before reaching the user.
+ */
+public class UnableToSetSerialFilterException extends Exception {
```

Comment:
Seems like several usage of this exception is to catch this and turn it into a RuntimeException. In my experience, checked exception are used when caller can do something about this exception. If caller can not do anything about this exception, would it be better to make this an unchecked exception?

Reply:
After using unchecked exceptions for internal-only exceptions resulted in some of these slipping out to customers, we came up with the rule that internal exceptions should be checked. It's ok for external User API exceptions to be unchecked though.